app/helpers/administrate/application_helper.rb

```diff
@@ -15,6 +15,20 @@ def render_field(field, locals = {})
       render locals: locals, partial: field.to_partial_path
     end
 
+    def requireness(field)
+      required_field?(field) ? "required" : "optional"
+    end
+
+    def required_field?(field)
+      has_presence_validator?(field.resource.class, field.attribute)
+    end
+
+    def has_presence_validator?(resource_class, field_name)
+      resource_class.validators_on(field_name).
```

Comment:
I don't know if using a trailing dot in a multi-line chain is a project standard but I'd go with https://github.com/rubocop-hq/ruby-style-guide#leading-dot-in-multi-line-chains.

Reply:
It is a project standard